src/main/java/org/checkerframework/specimin/UnsolvedClass.java

```diff
@@ -164,17 +164,29 @@ public void updateFieldByType(String currentType, String correctType) {
     Iterator<String> iterator = classFields.iterator();
     Set<String> newFields = new HashSet<>();
     while (iterator.hasNext()) {
-      List<String> elements = Splitter.on(' ').splitToList(iterator.next());
+      String fieldDeclared = iterator.next();
+      boolean isStatic = false;
+      if (fieldDeclared.startsWith("static")) {
+        fieldDeclared = fieldDeclared.replace("static ", "");
+        isStatic = true;
+      }
+      List<String> elements = Splitter.on(' ').splitToList(fieldDeclared);
       // fieldExpression is guaranteed to have the form "TYPE FIELD_NAME". Since this field
       // expression is from a synthetic class, there is no annotation involved, so TYPE has no
       // space.
       String fieldType = elements.get(0);
       String fieldName = elements.get(1);
       if (fieldType.equals(currentType)) {
         iterator.remove();
-        newFields.add(
-            UnsolvedSymbolVisitor.setInitialValueForVariableDeclaration(
-                correctType, correctType + " " + fieldName));
+        if (!isStatic) {
+          newFields.add(
+              UnsolvedSymbolVisitor.setInitialValueForVariableDeclaration(
+                  correctType, correctType + " " + fieldName));
+        } else {
+          newFields.add(
+              UnsolvedSymbolVisitor.setInitialValueForVariableDeclaration(
+                  correctType, "static " + correctType + " " + fieldName));
```

Comment:
Since these two calls are otherwise identical, I'd prefer to make `isStatic` a `String` variable rather than a boolean:
* if the field is static, then `isStatic` equals `"static "`
* if the field is non-static, then `isStatic` equals the empty string

We might want to rename the variable, too (maybe to just "static"?)